pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/UpsertTableIntegrationTest.java

```diff
@@ -245,14 +244,13 @@ protected void testDeleteWithFullUpsert(String kafkaTopicName, String tableName,
       }
     }, 100L, 600_000L, "Failed to load all upsert records for testDeleteWithFullUpsert");
 
-
     // Validate: pk is queryable and all columns are overwritten with new value
     rs = getPinotConnection()
         .execute("SELECT playerId, name, game FROM " + tableName + " WHERE playerId = 100").getResultSet(0);
     Assert.assertEquals(rs.getRowCount(), 1);
     Assert.assertEquals(rs.getInt(0, 0), 100);
     Assert.assertEquals(rs.getString(0, 1), "Zook-New");
-    Assert.assertEquals(rs.getString(0, 2), "null");
+    Assert.assertEquals(rs.getString(0, 2), "[\"null\"]");
```

Comment:
@shenyu0127 is this behavior change expected?

Reply:
This behavior is not expected.

Your changes to `PinotHelixResourceManager.java` are unrelated to ingestion. Why would the behavior change?